drivers/SmartThings/zigbee-window-treatment/src/aqara/version/init.lua

```diff
@@ -0,0 +1,77 @@
+local clusters = require "st.zigbee.zcl.clusters"
+local cluster_base = require "st.zigbee.cluster_base"
+local aqara_utils = require "aqara/aqara_utils"
+
+local Basic = clusters.Basic
+local WindowCovering = clusters.WindowCovering
+local AnalogOutput = clusters.AnalogOutput
+
+local INIT_STATE = "initState"
+local INIT_STATE_OPEN = "open"
+local INIT_STATE_CLOSE = "close"
+local INIT_STATE_REVERSE = "reverse"
+
+local APPLICATION_VERSION = "application_version"
+
+local function shade_level_report_legacy_handler(driver, device, value, zb_rx)
+  -- reported before initialized
+  aqara_utils.shade_position_changed(device, value)
+  aqara_utils.emit_shade_state_event(device, value.value)
+end
+
+local function shade_level_report_handler(driver, device, value, zb_rx)
+  -- emit shade event required
+  aqara_utils.shade_position_changed(device, value)
+  aqara_utils.emit_shade_state_event(device, value.value)
+end
+
+local function shade_state_report_handler(driver, device, value, zb_rx)
+  -- not reported in last initState step
+  aqara_utils.shade_state_changed(device, value)
+
+  -- initializedState
+  local state = value.value
+  if state == aqara_utils.SHADE_STATE_STOP or state == 0x04 then
```

Comment:
what's `4`? is this `4` the only different with the base handler?

Reply:
if this is the only difference, and this device is the only one that reports `4`, it can just be included in the base driver and you don't have to override the handler here. Does that makes sense? If the non-"version" device never send the value, it doesn't matter if this check is in the base handler or not